src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGitHubPlatform.java

```diff
@@ -0,0 +1,65 @@
+package com.sap.sgs.phosphor.fosstars.data.github;
+
+import static com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures.USES_GITHUB_PLATFORM;
+
+import com.sap.sgs.phosphor.fosstars.model.Value;
+import com.sap.sgs.phosphor.fosstars.model.ValueSet;
+import com.sap.sgs.phosphor.fosstars.model.value.BooleanValue;
+import com.sap.sgs.phosphor.fosstars.tool.github.GitHubProject;
+import java.io.IOException;
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+
+/**
+ * This data provider checks if a project uses GitHub as a main development platform. The check
+ * would be based on mirror url is null or empty. This will ensure whether the underlying project is
+ * a mirror project or not.
+ */
+public class UsesGitHubPlatform extends AbstractGitHubDataProvider {
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param github An interface to the GitHub API.
+   */
+  public UsesGitHubPlatform(GitHub github) {
+    super(github);
+  }
+
+  @Override
+  protected UsesGitHubPlatform doUpdate(GitHubProject project, ValueSet values) throws IOException {
+    logger.info("Figuring out if the project uses GitHub as the main development platform ...");
+
+    Optional<Value> something = cache.get(project, USES_GITHUB_PLATFORM);
+    if (something.isPresent()) {
+      values.update(something.get());
+      return this;
+    }
+
+    Value<Boolean> usesGitHubPlatform = usesGitHubPlatform(project);
+    values.update(usesGitHubPlatform);
+    cache.put(project, usesGitHubPlatform, tomorrow());
+
+    return this;
+  }
+
+  /**
+   * Checks if the GitHub repository has an empty or null mirror url.
+   *
+   * @param project The project.
+   * @return {@link BooleanValue} value of the feature.
+   */
+  private Value<Boolean> usesGitHubPlatform(GitHubProject project) {
+    // TODO: This is just a basic check, more constraints have to be considered in the future with
```

Comment:
If you have ideas what kind of check we can consider, could you please open an issue and put the details there?